src/main/python/systemml/defmatrix.py

```diff
@@ -116,50 +165,74 @@ def performDFS(outputs, execute):
 
-
-def binaryOp(lhs, rhs, opStr):
+def _log_base(val, base):
+    if not isinstance(val, str):
+        raise ValueError('The val to _log_base should be of type string')
+    return '(log(' + val + ')/log(' + str(base) + '))' 
+    
+def _matricize(lhs, inputs):
     """
-    Common function called by all the binary operators in matrix class
+    Utility fn to convert the supported types to matrix class or to string (if float or int)
+    and return the string to be passed to DML as well as inputs
     """
-    inputs = []
+    if isinstance(lhs, SUPPORTED_TYPES):
+        lhs = matrix(lhs)
     if isinstance(lhs, matrix):
         lhsStr = lhs.ID
-        inputs = [lhs]
+        inputs = inputs + [lhs]
     elif isinstance(lhs, float) or isinstance(lhs, int):
         lhsStr = str(lhs)
     else:
         raise TypeError('Incorrect type')
-    if isinstance(rhs, matrix):
-        rhsStr = rhs.ID
-        inputs = inputs + [rhs]
-    elif isinstance(rhs, float) or isinstance(rhs, int):
-        rhsStr = str(rhs)
-    else:
-        raise TypeError('Incorrect type')
-    return constructIntermediateNode(inputs, [OUTPUT_ID, ' = ', lhsStr, opStr, rhsStr, '\n'])
-
-def getValue(obj):
-    if isinstance(obj, matrix):
-        return obj.ID
-    elif isinstance(obj, float) or isinstance(obj, int):
-        return str(obj)
-    else:
-        raise TypeError('Unsupported type for ' + s)
+    return lhsStr, inputs
+    
+def binary_op(lhs, rhs, opStr):
+    """
+    Common function called by all the binary operators in matrix class
+    """
+    inputs = []
+    lhsStr, inputs = _matricize(lhs, inputs)
+    rhsStr, inputs = _matricize(rhs, inputs)
+    return construct_intermediate_node(inputs, [OUTPUT_ID, ' = ', lhsStr, opStr, rhsStr, '\n'])
 
 def binaryMatrixFunction(X, Y, fnName):
     """
     Common function called by supported PyDML built-in function that has two arguments.
     """
-    return constructIntermediateNode([X, Y], [OUTPUT_ID, ' = ', fnName,'(', getValue(X), ', ', getValue(Y), ')\n'])
+    inputs = []
+    lhsStr, inputs = _matricize(X, inputs)
+    rhsStr, inputs = _matricize(Y, inputs)
+    return construct_intermediate_node(inputs, [OUTPUT_ID, ' = ', fnName,'(', lhsStr, ', ', rhsStr, ')\n'])
 
 def unaryMatrixFunction(X, fnName):
     """
     Common function called by supported PyDML built-in function that has one argument.
     """
-    return constructIntermediateNode([X], [OUTPUT_ID, ' = ', fnName,'(', getValue(X), ')\n'])
+    inputs = []
+    lhsStr, inputs = _matricize(X, inputs)
+    return construct_intermediate_node(inputs, [OUTPUT_ID, ' = ', fnName,'(', lhsStr, ')\n'])
 
+def seq(start=None, stop=None, step=1):
+    """
+    Creates a single column vector with values starting from <start>, to <stop>, in increments of <step>.
+    Note: Unlike Numpy's arange which returns a row-vector, this returns a column vector.
+    Also, Unlike Numpy's arange which doesnot include stop, this method includes stop in the interval.
```

Comment:
I would prefer to only support the equivalent of `np.arange` in this Python DSL, rather than introduce our R-like `seq` function.  The users of the Python DSL will be Python users, so they will expect every aspect of our Python DSL to be Pythonic.